src/com/goide/inspections/GoPlaceholderChecker.java

```diff
@@ -0,0 +1,456 @@
+  private static final Map<String, Integer> FORMATTING_FUNCTIONS = ContainerUtil.newHashMap(
+    Pair.pair("errorf", 0),
+    Pair.pair("fatalf", 0),
+    Pair.pair("fprintf", 1),
+    Pair.pair("fscanf", 1),
+    Pair.pair("logf", 0),
+    Pair.pair("panicf", 0),
+    Pair.pair("printf", 0),
+    Pair.pair("scanf", 0),
+    Pair.pair("skipf", 0),
+    Pair.pair("sprintf", 0),
+    Pair.pair("sscanf", 1));
+
+  private static final List<String> PRINTING_FUNCTIONS = ContainerUtil.newArrayList(
+    "error",
+    "error",
+    "fatal",
+    "fprint",
+    "fprintln",
+    "log",
+    "panic",
+    "panicln",
+    "print",
+    "println",
+    "sprint",
+    "sprintln"
+  );
+
+  protected enum PrintfArgumentType {
+    ANY(-1),
+    BOOL(1),
+    INT(2),
+    RUNE(3),
+    STRING(4),
+    FLOAT(5),
+    COMPLEX(6),
+    POINTER(7);
+
+    private int myMask;
+
+    PrintfArgumentType(int mask) {
+      myMask = mask;
+    }
+
+    public int getValue() {
+      return myMask;
+    }
+  }
+
+  enum PrintVerb {
+    Percent('%', "", 0),
+    b('b', " -+.0", INT.getValue() | FLOAT.getValue() | COMPLEX.getValue()),
+    c('c', "-", RUNE.getValue() | INT.getValue()),
+    d('d', " -+.0", INT.getValue()),
+    e('e', " -+.0", FLOAT.getValue() | COMPLEX.getValue()),
+    E('E', " -+.0", FLOAT.getValue() | COMPLEX.getValue()),
+    f('f', " -+.0", FLOAT.getValue() | COMPLEX.getValue()),
+    F('F', " -+.0", FLOAT.getValue() | COMPLEX.getValue()),
+    g('g', " -+.0", FLOAT.getValue() | COMPLEX.getValue()),
+    G('G', " -+.0", FLOAT.getValue() | COMPLEX.getValue()),
+    o('o', " -+.0#", INT.getValue()),
+    p('p', "-#", POINTER.getValue()),
+    q('q', " -+.0#", RUNE.getValue() | INT.getValue() | STRING.getValue()),
+    s('s', " -+.0", STRING.getValue()),
+    t('t', "-", BOOL.getValue()),
+    T('T', "-", ANY.getValue()),
+    U('U', "-#", RUNE.getValue() | INT.getValue()),
+    V('v', " -+.0#", ANY.getValue()),
+    x('x', " -+.0#", RUNE.getValue() | INT.getValue() | STRING.getValue()),
+    X('X', " -+.0#", RUNE.getValue() | INT.getValue() | STRING.getValue());
+
+    private char myVerb;
+    private String myFlags;
+    private int myMask;
+
+    PrintVerb(char verb, String flags, int mask) {
+      myVerb = verb;
+      myFlags = flags;
+      myMask = mask;
+    }
+
+    public char getVerb() {
+      return myVerb;
+    }
+
+    @NotNull
+    public String getFlags() {
+      return myFlags;
+    }
+
+    public int getMask() {
+      return myMask;
+    }
+
+    @Nullable
+    public static PrintVerb getByVerb(char verb) {
+      for (PrintVerb v : values()) {
+        if (verb == v.getVerb()) return v;
+      }
+      return null;
+    }
+  }
+
+  public static boolean isFormattingFunction(String functionName) {
+    return FORMATTING_FUNCTIONS.containsKey(functionName);
+  }
+
+  public static boolean isPrintingFunction(String functionName) {
+    return PRINTING_FUNCTIONS.contains(functionName);
```

Comment:
I've tried `indexOf()` but the IDEA will say I should use `contains()`. Should I use a simple `for` instead?